ruby_oop/mastermind/game.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require './computer_player'
+require './human_player'
+
+# class to play the mastermind game
+class Game
+  attr_accessor :human_player, :computer_player
+
+  AVAILABLE_COLORS = ['🟤', '🔴', '🔵', '🟢', '🟡', '🟣'].freeze
+  @secret_code = []
+
+  def initialize
+    @human_player = codebreaker_or_codemaker
+    @computer_player = !@human_player
+  end
+
+  def computer_codemaker
+    puts "Computer is selecting random colors...\n\n"
+    @secret_code = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@secret_code)
+  end
+
+  def computer_codebreaker
+    puts "Computer is selecting random colors...\n\n"
+    @guess_array = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@guess_array)
+  end
+
+  def human_codemaker
+    puts 'Enter a code to be broken: '
+    @secret_code = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+      HumanPlayer.new(@secret_code)
+  end
+
+  def human_codebreaker
+    @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                    pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    if @guess_array.include?(nil) || @guess_array.uniq.length != 4
+      puts 'Duplicate entries or empty entries not allowed. Enter a new selection again.'
+      @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    end
+    puts "\nYou guessed: #{@guess_array}"
+    HumanPlayer.new(@guess_array)
```

Comment:
I see here and on line 34 that you are instantiating new `HumanPlayer`s.   To me it doesn't seem like this is being used, though.  The `HumanPlayer` is not stored in a `Game` class variable and I don't see you calling the `HumanPlayer.secret_code` anywhere.

The same looks to be true for `ComputerPlayer`.  Given your implementation I think you could do away with those classes.  Happy to pair on this one if needed, I know I started you down this path 😬

Reply:
I struggled with this too, and agree with Kevin. This is worth re-thinking through IMO.